app/services/google_service.rb

```diff
@@ -0,0 +1,32 @@
+class GoogleService
+    attr_reader :id
+
+    def initialize(location)
+        @id = 11
+        @location = get_json(location)
+    end
+
+    def latitude
+        @location[:results][0][:geometry][:location][:lat]
+    end
+
+    def longitude
+        @location[:results][0][:geometry][:location][:lng]
+    end
+
+    private
+    attr_reader :location
+
+    def get_json(location)
+        response = connection(location).get("maps/api/geocode/json")
+        JSON.parse(response.body, symbolize_names: true)
+    end
+
+    def connection(location)
+        Faraday.new(
+            url: 'https://maps.googleapis.com/',
+            params: { key: ENV['GOOGLE_API_KEY']
+                    }.merge({address: location['address']})
```

Comment:
Line 29 is interesting... could you explain what is going on here? Why is the address in a hash?

Reply:
I referenced an older Faraday connection to the same Google Geocode API. For the Geocode API, you must pass in both the API key as well as the location you want coordinates for. Originally, it was .merge(location) but since we are passing in private params. it was a params object. To correct, I created an actual hash/object with the address key and private params address as the pair.

Reply:
Ah great, I understand now. Thank you for explaining that! It just looked a bit odd to me.